File: detrital-wasserstein-1.0/w2.R
# Load in the packages
library(transport)
# Load data
vefsna <- read.csv("vefsna.csv", header = TRUE, sep = ",")
vefsna <- as.vector(vefsna[[1]])
byskealven <- read.csv("byskealven.csv", header = TRUE, sep = ",")
byskealven <- as.vector(byskealven[[1]])
# Calculate W_p between vefsna and byskealven samples, for p=2
w2 <- wasserstein1d(vefsna, byskealven, p = 2)
cat("W2 =", w2, "\n")

File: detrital-wasserstein-1.0/w2.py
# Load in the packages
import numpy as np
import ot

# Load data
vefsna = np.loadtxt("vefsna.csv", delimiter=",", skiprows=1)
byskealven = np.loadtxt("byskealven.csv", delimiter=",", skiprows=1)
# Calculate W_p^p between vefsna and byskealven samples, for p=2
W2_2 = ot.wasserstein_1d(vefsna, byskealven, p=2)
# Calculate W2 using square root
W2 = np.sqrt(W2_2)
print("W2 =", W2)
